app.py

```diff
@@ -55,55 +76,128 @@ def _get_analyzers():
         }
     )
     response.raise_for_status()
-    _LOGGER.debug('Full response from Kubernetes master is: %r', response.json())
+    _LOGGER.debug(f"Full response from Kubernetes master is: {response.json()}")
+
     pods = response.json().get('items', [])
-    _LOGGER.info('Kubernetes master returned %d pods', len(pods))
+    _LOGGER.info(f"Kubernetes master returned {len(pods)} pods with label {label_selector}")
+
     return pods
 
 
+def _get_jobs(label_selector: str) -> dict:
+    """Get succeeded jobs by label."""
+    # TODO: pagination?
+    # TODO: if label_selector is empty do not use it for requests.get()
+    endpoint = (f"{KUBERNETES_API_URL}/apis/batch/v1/namespaces/{THOTH_MIDDLETIER_NAMESPACE}/"
+                f"jobs?labelSelector={label_selector}&limit=500")  # we play nice and just get 500
+
+    response = requests.get(
+        endpoint,
+        verify=bool(int(os.getenv('KUBERNETES_VERIFY_TLS', True))),
+        headers={
+            'Authorization': 'Bearer {}'.format(KUBERNETES_API_TOKEN),
+            'Content-Type': 'application/json'
+        }
+    )
+    response.raise_for_status()
+    _LOGGER.debug(f"Full response from Kubernetes master is: {response.json()}")
+
+    jobs = response.json().get('items', [])
+    _LOGGER.info(f"Kubernetes master returned {len(jobs)} jobs with label {label_selector}")
+
+    return jobs
+
+
 def _delete_pod(pod_name):
-    endpoint = "{}/{}/{}/{}".format(KUBERNETES_API_URL,
-                                    'api/v1/namespaces',
-                                    THOTH_MIDDLETIER_NAMESPACE,
-                                    'pods',
-                                    pod_name)
+    """Delete a named pod."""
+    endpoint = f"{KUBERNETES_API_URL}/api/v1/namespaces/{THOTH_MIDDLETIER_NAMESPACE}/{pod_name}"
+
     response = requests.delete(
         endpoint,
-        verify=bool(os.getenv('KUBERNETES_VERIFY_TLS', True)),
+        verify=KUBERNETES_VERIFY_TLS,
         headers={
             'Authorization': 'Bearer {}'.format(KUBERNETES_API_TOKEN),
             'Content-Type': 'application/json'
         }
     )
     response.raise_for_status()
-    _LOGGER.debug('Response from Kubernetes master on deletion is: %r', response.json())
+    _LOGGER.debug('Response from Kubernetes master on deletion is: {response.json()}')
```

Comment:
You forgot ``f`` in f-string. Anyway, this is considered as bad practice since there is constructed whole string regardless of log level (if logger is for example INFO, this string will be constructed, passed to the method ``debug`` and thrown away). Passing it as additional arguments for string expansion saves necessary string construction due to debug level.

Reply:
uh, nice... I just tried to unform all the string stuff....